### PR TITLE
Proposed Q1 2020 dioterms updates

### PR DESCRIPTION
**Pull request by yesnet0**
- VDP and BBP terms separated to reduce ambiguity
- Terms split into modules for easier language and legal translation and to support a web policy creation front-end
- Safeharbor language modified to reflect commonly-used language seen in 2H20
- Some semantic and language tweaks in the README and elsewhere
- Introduction of Disclose.io Status

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +28,20 @@ The intention of Safe Harbor language provided by disclose.io is for it to be fo
 
 Policies missing any of the core tenets above, but that still contain a good-faith statement committing not to pursue legal action on security researchers, meet the criteria for **Partial Safe Harbor**.
 
+### Disclosure types
+
+- **Coordinated Disclosure**: A researcher can share details of the vulnerability after a fix has been applied and the program owner has provided permission to disclose or after a clearly-stated time has passed from submission, whichever is sooner;
+- **Discretionary Disclosure**: The researcher or the program owner can request mutual permission to share details of the vulnerability after approval is explicitly received; or
```
Comment (haasonsaas) [inline README.md:36]: Is it worth providing that this may be something that is a process that can happen in whole or in part?

File: core-modules/generic-core-terms-03-our-commitments.md.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## Our Commitments
+
+When working with us, according to this policy, you can expect us to:
+
+- Respond to your report promptly, and work with you to understand and validate your report 
+- Strive to keep you informed about the progress of a vulnerability as it is processed
+- Work to remediate discovered vulnerabilities in a timely manner
+- Extend Safe Harbor for your vulnerability research that is related to this policy
```
Comment (haasonsaas) [inline core-modules/generic-core-terms-03-our-commitments.md.md]: Presuming it falls within the bounds as explained within? Or is that implicit here?

File: core-modules/generic-core-terms-04-our-expectations.md.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+- Play by the rules, including following this policy and any other relevant agreements. If there is any inconsistency between this policy and any other applicable terms, the terms of this policy will prevail;
+- Report any vulnerability you’ve discovered promptly;
+- Avoid violating the privacy of others, disrupting our systems, destroying data, and/or harming user experience;
+- Use only the Official Channels to discuss vulnerability information with us;
+- Handle the confidentiality of details of any discovered vulnerabilities according to our Disclosure Policy;
+- Perform testing only on in-scope systems, and respect systems and activities which are out-of-scope;
+- If a vulnerability provides unintended access to data: Limit the amount of data you access to the minimum required for effectively demonstrating a Proof of Concept; and cease testing and submit a report immediately if you encounter any user data during testing, such as Personally Identifiable Information (PII), Personal Healthcare Information (PHI), credit card data, or proprietary information;
+- You should only interact with test accounts you own or with explicit permission from the account holder; and
```
Comment (haasonsaas) [inline core-modules/generic-core-terms-04-our-expectations.md.md]: What about in scenarios in which the holder is unclear, or potentially is not the party which can/should be capable of granting such access? (e.g. employee providing a company account to a researcher)

**Review by haasonsaas — COMMENTED** [review]
A few ?'s

File: README.md (hunk 1 above)
Comment (yesnet0) [reply README.md:36]: I'm not totally following - Can you give an example?

In general, the language prefers strict definitions in order to avoid ambiguity.

File: core-modules/generic-core-terms-03-our-commitments.md.md (hunk 2 above)
Comment (yesnet0) [reply core-modules/generic-core-terms-03-our-commitments.md.md]: Great question - It's implicit

File: core-modules/generic-core-terms-04-our-expectations.md.md (hunk 3 above)
Comment (yesnet0) [reply core-modules/generic-core-terms-04-our-expectations.md.md]: Mmm... That's somewhat of an edge case. I'd think that's more of an operational consideration than one that could be covered by the policy

File: README.md [hunk 4]
```diff
@@ -22,6 +28,20 @@ The intention of Safe Harbor language provided by disclose.io is for it to be fo
 
 Policies missing any of the core tenets above, but that still contain a good-faith statement committing not to pursue legal action on security researchers, meet the criteria for **Partial Safe Harbor**.
 
+### Disclosure types
+
```
Comment (CharlieEriksen) [inline README.md:34]: Is it worth calling out here that these are not an either/or with VDP/BBPs? But rather, you can have a BBP or VDP together with one of these disclosure types?

File: core-terms-vdp.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# [organization] Vulnerability Disclosure Policy
+
+## Introduction
+
+[organization] welcomes feedback from security researchers and the general public to help improve our security. If you believe you have discovered a vulnerability, privacy issue, exposed data, or other security issues in any of our assets, we want to hear from you. This policy outlines steps for reporting vulnerabilities to us, what we expect, what you can expect from us.
+
+## Scope
+
+This policy applies to any digital assets owned, operated, or maintained by [organization], including public facing websites.
+
```
Comment (dantrauner) [inline core-terms-vdp.md]: Looking at https://github.com/disclose/dioterms/pull/5/files#diff-d49350228ef1f6c3dc92a1a3ddf500d56a685ee18a0f9631fe5066d206711482R7-R12, it seems like the idea is that a "Bug Bounty Program" has to define custom in/out of scope assets and rewards, while a VDP should use this specific language for its "Scope" section?

I wasn't sure if this implied that a VDP always needed to be relevant to "any digital assets owned, operated, or maintained" by the org. I can see cases where you want a "scoped" VDP in some sense – e.g. if you sell an on-prem product and want to scope the program to instances of the on-prem product that the researcher has permission to test. There are other edge cases like third-party SaaS services purchased by the VDP owner where they might have some power to modify the configuration of their tenant, but may not be able to authorize certain types of testing against that web infrastructure as it's not actually owned by the VDP owner.

Comment (yesnet0) [reply core-terms-vdp.md]: That's correct. The intent here would be to more clearly reflect and reinforce that a VDP ultimately doesn't create control over any testing a researcher does (or, indeed, where a security issue may be identified by someone who isn't doing proactive security research). In contrast, a bounty program may have incentives in one area and not in others (cc: @cablej)

File: README.md (hunk 4 above)
Comment (yesnet0) [reply README.md:34]: Interesting idea. How would you articulate that distinction more clearly in the language? Remember, one of the goals is to minimize words where possible (but I also agree with your statement).

File: core-terms-vdp.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+## Our Expectations
+
+In participating in our vulnerability disclosure program in good faith, we ask that you:
+
+- Play by the rules, including following this policy and any other relevant agreements. If there is any inconsistency between this policy and any other applicable terms, the terms of this policy will prevail;
+- Report any vulnerability you’ve discovered promptly;
+- Avoid violating the privacy of others, disrupting our systems, destroying data, and/or harming user experience;
+- Use only the Official Channels to discuss vulnerability information with us;
+- Handle the confidentiality of details of any discovered vulnerabilities according to our Disclosure Policy;
```
Comment (cablej) [inline core-terms-vdp.md]: Just because I could see people copy-and-pasting these terms, and it might not be clear that Disclosure Policy is something separate, thoughts on adding a default disclosure policy of 90 days? This would have the added benefit of driving more organizations to 90 days and showing that it is an industry standard.

Sample language:

> Keep the details of any discovered vulnerabilities confidential until either they are fixed or at least 90 days have passed.

File: README.md (hunk 1 above)
Comment (haasonsaas) [reply README.md:36]: Sure, so this would be in an instance where a vulnerability disclosure contains certain privileged information that for whatever reason is not ready for public consumption (e.g. you can disclose X if you omit detail Y). Would this fall within this category?

File: core-terms-vdp.md (hunk 6 above)
Comment (yesnet0) [reply core-terms-vdp.md]: @cablej I'd like to work towards this, but am concerned that it creates too much friction for orgs considering whether or not they should do a VDP in the first place (this decision usually comes before a decision to set a proactive timeline). The Disclose.io Status including in this PR introduces a maturity model where Safeharbor plus CVD is the "three gold star - you're the awesomest" variant of a VDP policy to encourage orgs towards CVD... Do you think that's enough, or would providing timelines boilerplates help here as well?

Comment (yesnet0) [reply core-terms-vdp.md]: @cablej The other challenge which needs to be solved is the varying timeline requirements of products - 90 days is perfectly reasonable imo for SaaS platforms and widely-deployed on-premise software, but mightn't be suitable for a voting machine company, medical device manufacturer, satellite equipment company, car company etc etc etc where the regression testing and in-the-wild patch deployment cycles are potentially longer. Any thoughts on how to address these variations are definitely appreciated - It's an easy conversation to have with organizations 1-to-1 but a difficult one to assert a template on.

File: README.md (hunk 1 above)
Comment (yesnet0) [reply README.md:36]: Yes, in coordinated and discretionary disclose - provided that the communication channel between the finder and recipient is active - it's customary for there to be a conversation about what is wise to include or omit from public disclosure.

File: README.md (hunk 4 above)
Comment (yesnet0) [reply README.md:34]: @charlieeriksen A thought here would be that public open-scope bug bounty programs are unlikely to coexist with an open-scope VDP - Are you referring to closed-scope or private programs here?

File: archive/generic-core-terms.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# Expectations
+
+When working with us according to this policy, you can expect us to:
+
+- Work with you to understand and validate your report, including a timely initial response to the submission;
+- Work to remediate discovered vulnerabilities in a timely manner; and
+- Recognize your contribution to improving our security if you are the first to report a unique vulnerability, and your report triggers a code or configuration change.
+
+# Scope
+
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion: 

The scope of this policy includes only assets owned by the party participating in this policy, specifically:
[...]

Any systems not listed above are out-of-scope for security testing under this policy. Any discovered or suspected vulnerabilities in out of scope systems should be reported to the appropriate vendor or applicable authority.

File: archive/generic-core-terms.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+# Out-of-Scope
+
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion:

Assets or other equipment not owned by parties participating in this policy or not listed in "Systems In Scope."

File: archive/generic-core-terms.md (hunk 7 above)
Comment (dantrauner) [reply archive/generic-core-terms.md]: Regarding this, we were actually looking at adjusting the language for our own purposes (for VDP, which is designed to be open scope – [here](https://github.com/disclose/dioterms/pull/5/files#diff-5b6df7121eb9ca278274a6d72cb436a5b1617322faa8416233e913feb125c01fR9)) to something like:

> This policy applies to any digital assets owned, operated, or maintained by $COMPANY for which $COMPANY can legally authorize the testing of, including instances of the $COMPANY product which you have explicit permission to test.

File: archive/generic-core-terms.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
+# Introduction
+
+Security is core to our values, and we value the input of hackers acting in good-faith to help us maintain a high standard for the security and privacy for our users. This includes encouraging responsible vulnerability research and disclosure. This policy sets out our definition of good-faith in the context of finding and reporting vulnerabilities, as well as what you can expect from us in return.
+
+# Expectations
+
+When working with us according to this policy, you can expect us to:
+
+- Work with you to understand and validate your report, including a timely initial response to the submission;
+- Work to remediate discovered vulnerabilities in a timely manner; and
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested edit:

in a timely manner _within our operational constraints_;

File: archive/generic-core-terms.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
+# Introduction
+
+Security is core to our values, and we value the input of hackers acting in good-faith to help us maintain a high standard for the security and privacy for our users. This includes encouraging responsible vulnerability research and disclosure. This policy sets out our definition of good-faith in the context of finding and reporting vulnerabilities, as well as what you can expect from us in return.
+
+# Expectations
+
+When working with us according to this policy, you can expect us to:
+
+- Work with you to understand and validate your report, including a timely initial response to the submission;
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion:

- Prioritize the security and privacy of our users;

File: archive/generic-core-terms.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+- Work to remediate discovered vulnerabilities in a timely manner; and
+- Recognize your contribution to improving our security if you are the first to report a unique vulnerability, and your report triggers a code or configuration change.
+
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion: 

While we will always strongly consider your assessment and recommendations regarding vulnerability severity, we retain the authority to determine what vulnerabilities can and should be remediated and in what time frame.

File: archive/generic-core-terms.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+- Play by the rules. This includes following this policy any other relevant agreements;
+- Report any vulnerability you’ve discovered promptly;
+- Avoid violating the privacy of others, disrupting our systems, destroying data, and/or harming user experience;
+- Use only the Official Channels to discuss vulnerability information with us;
+- Handle the confidentiality of details of any discovered vulnerabilities according to our Disclosure Policy;
+- Perform testing only on in-scope systems, and respect systems and activities which are out-of-scope;
+- If a vulnerability provides unintended access to data: Limit the amount of data you access to the minimum required for effectively demonstrating a Proof of Concept; and cease testing and submit a report immediately if you encounter any user data during testing, such as Personally Identifiable Information (PII), Personal Healthcare Information (PHI), credit card data, or proprietary information;
+- You should only interact with test accounts you own or with explicit permission from the account holder; and
+- Do not engage in extortion.
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested edit:

- Do not engage in extortion _or phishing. Do not misrepresent yourself to be an employee or government official_.

File: archive/generic-core-terms.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion:

- Do not exfiltrate, modify, or destroy system data.

File: archive/generic-core-terms.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+# Safe Harbor
+
+When conducting vulnerability research according to this policy, we consider this research conducted under this policy to be:
+
+- Authorized in view of any applicable anti-hacking laws, and we will not initiate or support legal action against you for accidental, good faith violations of this policy;
+- Authorized in view of relevant anti-circumvention laws, and we will not bring a claim against you for circumvention of technology controls;
+- Exempt from restrictions in our Acceptable Usage Policy that would interfere with conducting security research, and we waive those restrictions on a limited basis; and
+- Lawful, helpful to the overall security of the Internet, and conducted in good faith.
+
+You are expected, as always, to comply with all applicable laws. If legal action is initiated by a third party against you and you have complied with this policy, we will take steps to make it known that your actions were conducted in compliance with this policy.
```
Comment (HarleyGeiger) [inline archive/generic-core-terms.md]: Suggested insertion: 

Please note that the safe harbor applies only to legal claims under the control of the organization participating in this policy, and that the policy does not bind independent third parties.

File: core-terms-vdp.md (hunk 6 above)
Comment (HarleyGeiger) [reply core-terms-vdp.md]: While I'd like to see faster vuln remediation, a 90-day deadline is not feasible for many situations or organizations. Trying to impose a 90-day deadline on disclose.io users seems to be separate from the goal of broadening VDP adoption.

File: README.md [hunk 15]
```diff
@@ -31,18 +50,12 @@ In each template we've also provided boilerplate examples for the additional sec
...
-- **Discretionary Disclosure**: The researcher or the program owner can request mutual permission to share details of the vulnerability after approval is explicitly received; or
-- **Non-Disclosure**: Researchers are required to keep vulnerability details and the existence of the program itself confidential.
-
 ## Next steps 
 
 Once you've published your policy, you can:  
 - Add the appropriate [disclose.io seal](https://github.com/disclose/dioseal) to your public program brief,
 - Submit a pull request to add your program to the [open-source disclose.io program list](https://github.com/disclose/diodb),
-- Let the world know you're joining the movement!
+- Let the world know you're joining the initiative!
```
Comment (w8mej) [inline README.md:61]: Feels more akin to a coalition.  An alliance for combined action on VDP / BBP efforts.  My personal experience is that one forms a coalition in response to opportunities and threats.

File: archive/generic-core-terms.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
+# Introduction
+
+Security is core to our values, and we value the input of hackers acting in good-faith to help us maintain a high standard for the security and privacy for our users. This includes encouraging responsible vulnerability research and disclosure. This policy sets out our definition of good-faith in the context of finding and reporting vulnerabilities, as well as what you can expect from us in return.
```
Comment (w8mej) [inline archive/generic-core-terms.md]: Remove the run on - '..., and...' after values.  Should be: Security is core to our values.  We value.....

Comment (w8mej) [reply archive/generic-core-terms.md]: Not sure how to make more succinct.   "...This includes encouraging responsible vulnerability research and disclosure. This policy sets out our definition of good-faith in the context of finding and reporting vulnerabilities, as well as what you can expect from us in return..."  Maybe rip it all out.  Then put in the item below: 

Something akin to the following - "....This policy sets out our definition of good-faith in the context of responsible vulnerability research and disclosure, as well as set expectations on what to expect in return....."

File: archive/generic-core-terms.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+# Out-of-Scope
+
+# Rewards
+
+# Disclosure Policy
+
+# Ground Rules
+
+To encourage vulnerability research and to avoid any confusion between legitimate research and malicious attack, we ask that you attempt, in good faith, to:
```
Comment (w8mej) [inline archive/generic-core-terms.md]: Rephrase "malicious attack" with something akin to "unauthorized access" ? "illegal activities" ? "unauthorized activities" ? "illegitimate activities" ?

**Review by w8mej — COMMENTED** [review]
Took a stab.

File: README.md (hunk 15 above)
Comment (yesnet0) [reply README.md:61]: I'm a big fan of this... Good add.

File: archive/generic-core-terms.md (hunk 16 above)
Comment (yesnet0) [reply archive/generic-core-terms.md]: @cloudriseup Curious re the second recommendation here: What's the rationale behind that?

File: archive/generic-core-terms.md (hunk 17 above)
Comment (yesnet0) [reply archive/generic-core-terms.md]: I'm a fan of "illegal activities" as a replacement here, although the ambiguity between what is legal and illegal in the first place makes it problematic and is why we steered clear of legal/illegal originally. Interested in the input of others here.

File: core-terms-vdp.md (hunk 6 above)
Comment (yesnet0) [reply core-terms-vdp.md]: @harleygeiger Perhaps putting different options in there could work as an approach? Alternatively, we leave this as is, make the other repo changes, and kick off a draft for improving the variations on disclosure terms

File: archive/generic-core-terms.md (hunk 7 above)
Comment (yesnet0) [reply archive/generic-core-terms.md]: @harleygeiger - Would @dantrauner's language work here? The only challenge I see is the tendency for organizations not to know to 100% which assets on the internet are theirs

Comment (cablej) [reply archive/generic-core-terms.md]: I would lean towards keeping terms general for a VDP, since it may not always be possible for an organization to enumerate their assets. 

Perhaps:

This policy applies to any assets owned, operated, or maintained by [organization], including the following:

...

If you are unsure whether a system belongs to us, please reach out before testing. Any discovered or suspected vulnerabilities in out of scope systems should be reported to the appropriate vendor or applicable authority.

File: core-terms-vdp.md (hunk 6 above)
Comment (cablej) [reply core-terms-vdp.md]: Maybe go the route of CISA's directive and state `Provide us a reasonable amount of time to resolve the issue before you disclose it publicly.`

File: README.md (hunk 4 above)
Comment (HarleyGeiger) [reply README.md:34]: Policies missing any of the core tenets above, but that still contain a good-faith statement committing not to pursue legal action on security researchers, _may_ meet the criteria for Partial Safe Harbor. _Incentives or "bounties" for vulnerability disclosures are not a prerequisite._

File: core-terms-vdp.md (hunk 6 above)
Comment (HarleyGeiger) [reply core-terms-vdp.md]: Keep the details of any discovered vulnerabilities confidential _while we are assessing and/or mitigating the vulnerabilities_, until either they are _addressed or an unreasonable amount of time has passed without progress_.

File: archive/generic-core-terms.md (hunk 7 above)
Comment (HarleyGeiger) [reply archive/generic-core-terms.md]: > @HarleyGeiger - Would @dantrauner's language work here? The only challenge I see is the tendency for organizations not to know to 100% which assets on the internet are theirs

Maybe combine them:

This policy applies only to any digital assets owned, operated, or maintained by $COMPANY for which $COMPANY can legally authorize the testing of. Any assets not listed above are out-of-scope for security testing under this policy. Vulnerabilities discovered or suspected in out-of-scope systems should be reported to the appropriate vendor or applicable authority.

File: archive/generic-core-terms.md (hunk 17 above)
Comment (HarleyGeiger) [reply archive/generic-core-terms.md]: Malicious attacks are almost always illegal and not legitimate research. But legitimate research can still sometimes be an illegal activity. It's splitting hairs to some extent, but my vote is to stick with "malicious attack."

File: archive/generic-core-terms.md (hunk 12 above)
Comment (yesnet0) [reply archive/generic-core-terms.md]: Going to nix "employee" from this on account of confused deputy attacks and the like. The government official line feels adequate imo.

**Review by codingo — APPROVED** [review]
